File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); i++)
        {
            int key = target - nums[i];
            if (m.find(key) != m.end())
            {
                return {m[key], i};
            }

            if (m.find(nums[i]) == m.end())
            {
                m[nums[i]] = i;
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/counting-bits/submission-1.java
class Solution {
    public int[] countBits(int n) {
        var result = new int[n + 1];
        result[0] = 0;

        if (n == 0)
            return result;
        result[1] = 1;

        for (int i = 2; i < n + 1; i++) {

            result[i] = result[i & i - 1] + 1;
        }

        return result;
    }
}
